src/containers/bookmark/Bookmark.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Icon } from 'antd';
+import { bindActionCreators } from 'redux';
+import { getBookmarks, addBookmark, removeBookmark } from '../../actions/bookmarks';
+import './bookmark.scss';
+
+export class UnconnectedBookmark extends Component {
+  componentWillMount() {
+    const { getBookmarks } = this.props;
+    getBookmarks();
+  }
+
+  checkBookmark = (bookmarks) => {
+    const { addBookmark, slug, removeBookmark } = this.props;
+    let bookmarkIcon = (
+      <Icon
+        type="book"
+        className="bookmark-container-add-icon"
+        data-test="bookmark-container-add-icon"
+        onClick={() => addBookmark(slug)}
+      />
+    );
+
+    bookmarks.forEach(article => (
+      article.slug === slug
+      && (
+        bookmarkIcon = (
+          <Icon
+            type="book"
+            className="bookmark-container-remove-icon"
+            data-test="bookmark-container-remove"
+            onClick={() => removeBookmark(slug)}
+          />
+        )
+      )
+    ));
+    return bookmarkIcon;
+  }
+
+  displayBookmark = () => {
+    const { bookmarks, slug, addBookmark } = this.props;
+    return Object.getOwnPropertyNames(bookmarks).length !== 0
+      && (
+        bookmarks.message
+          ? (
+            <Icon
+              type="book"
+              className="bookmark-container-add-icon"
+              data-test="bookmark-container-add"
+              onClick={() => addBookmark(slug)}
+            />
+          )
+          : this.checkBookmark(bookmarks['bookmarked articles'])
+      );
+  }
+
+  render() {
+    return (
+      <div
+        data-test="bookmark-container"
+        className="bookmark-container"
+      >
+        {this.displayBookmark()}
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state) => {
```

Comment:
`const mapStateToProps = state => {...}` should work just fine here. The parenthesis are not needed if you're only passing one parameter.

Reply:
At this point, how many items are in state?

Reply:
I have changed it to 
```
const mapStateToProps = ({ bookmarks }) => ({
  bookmarks,
});
```